BEST_PRACTICES.md

```diff
@@ -203,3 +216,459 @@ Use a simple NSURLSession wrapper to make things simpler, [Networking](https://g
 
 Old style layout is still an option for highly dynamic layouts such as when subclassing UICollectionViewLayout, but in general lines, first try Auto Layout.
+
+# Swift Style Guide
+
+Our overarching goals are conciseness, readability, and simplicity.
+
+## Naming
+
+Use descriptive names with camel case for classes, methods, variables, etc. Class names should be capitalized, while method names, variables and constants should start with a lower case letter.
+
+**Preferred:**
+
+```swift
+class WidgetContainer {
+    var widgetButton: UIButton
+    let widgetHeightPercentage = 0.85
+}
+```
+
+**Not Preferred:**
+
+```swift
+let MAX_WIDGET_COUNT = 100
+
+class app_widgetContainer {
+    var wBut: UIButton
+    let wHeightPct = 0.85
+}
+```
+
+For functions and init methods, prefer named parameters for all arguments.
+
+Declarations:
+
+```swift
+class Guideline {
+    func combineWithString(incoming incoming: String, options: Dictionary?)
+    func upvoteBy(amount amount: Int)
+    func date(string string: String) -> NSDate
+    func timedAction(delay: NSTimeInterval, action: SKAction) -> SKAction!
+    func authenticate(username username: String, password: String, completion: (error: NSError?) -> Void)
+}
+```
+
+Implementations:
+
+```swift
+date(string: "2014-03-14")
+
+timedAction(delay: 1.0, action: someOtherAction) {
+
+}
+
+authenticate(username: "elvis", password: "m8eu201je") { error in
+
+}
+```
+
+When in doubt, look at how Xcode lists the method in the jump bar – our style here matches that.
+
+![Methods in Xcode jump bar](https://raw.githubusercontent.com/raywenderlich/swift-style-guide/master/screens/xcode-jump-bar.png)
+
+### Class Prefixes
+
+Swift types are all automatically namespaced by the module that contains them. As a result, prefixes are not required in order to minimize naming collisions. If two names from different modules collide you can disambiguate by prefixing the type name with the module name:
+
+```swift
+import MyModule
+
+var myClass = MyModule.MyClass()
+```
+
+You **should not** add prefixes to your Swift types.
+
+## Spacing and Indentation
+
+* Indent using 4 spaces rather than tabs. This should be configured on the project.
+
+  ![Xcode indent settings](https://raw.githubusercontent.com/bakkenbaeck/iOS-playbook/master/assets/xcode-text-settings-swift.png)
+
+* Avoid doing method indentation since Swift's indentation is inconsistent, and just keep things in one line as long as is possible. It helps you to not having to think many times on how you would indent your code.
+
+**Preferred:**
+```swift
+let user = User(name: "Igor", lastName: "Ranieri", country: "Germany")
+```
+
+**Not Preferred:**
+```swift
+let user = User(name: "Igor",
+      lastName: "Ranieri",
+      country: "Germany")
+```
+
+* Method braces and other braces (`if`/`else`/`switch`/`while` etc.) always open on the same line as the statement but close on a new line.
```

Comment:
we also don't use parentheses for `if`, and by @!&*@# don't do `if()` as if it were a function? 💃